packaging/utils.py

```diff
@@ -115,14 +115,16 @@ def parse_wheel_filename(
 
 
 def parse_sdist_filename(filename: str) -> Tuple[NormalizedName, Version]:
-    if not filename.endswith(".tar.gz"):
+    match = re.match(r".*(\.tar\.gz|\.zip)$", filename)
```

Comment:
I’d just do `not any(filename.endswith(suffix) for suffix in (".tar.gz", ".zip"))`, no need to use regex for this (it’s probably slower as well).

Reply:
I'm using the regex to find the length of the extension as well. This is used in parsing the file name on line 127. I can switch to hard coded extensions lengths though if you prefer.

Reply:
I think something like this would be fine too:

```
if filename.endswith(".tar.gz"):
    name_without_extension = filename[:-len(".tar.gz")]
elif [same, thing, for, zip]:
     ...
else:
     raise

name_part, sep, version_part = name_without_extension.rpartition("-")
```

Reply:
Agreed I think @pradyunsg's approach keeps things simple.

Reply:
Thanks for the feedback. PR updated.

Reply:
You all are making me really want to get us to drop support up to Python 3.9 for `str.removesuffix()`. 😄

Reply:
_one day_ 🌈

Reply:
But first, we should probably make a PyPI release dropping Python 2.7. :P